veros/core/external/solvers/scipy_jax.py

```diff
@@ -1,19 +1,24 @@
 from veros import distributed, veros_routine, veros_kernel, runtime_state as rst
 from veros.variables import allocate
 
-from veros.core import utilities
 from veros.core.operators import update, update_add, at, numpy as npx
 from veros.core.external.solvers.base import LinearSolver
+from veros.core.external.poisson_matrix import assemble_poisson_matrix
 
 
 @veros_kernel(static_args=("solve_fun",))
 def solve_kernel(state, rhs, x0, boundary_val, solve_fun):
     vs = state.variables
+    settings = state.settings
 
     rhs_global = distributed.gather(rhs, state.dimensions, ("xt", "yt"))
     x0_global = distributed.gather(x0, state.dimensions, ("xt", "yt"))
 
-    boundary_mask = ~npx.any(vs.boundary_mask, axis=2)
+    if settings.enable_streamfunction:
+        boundary_mask = ~npx.any(vs.boundary_mask, axis=2)
+    else:
+        boundary_mask = vs.maskT[..., -1]
```

Comment:
This occurs way to many times throughout the code base. Can you come up with a way to handle this better? Should we remove the `boundary_mask` variable, or add a new one, or change the semantics?